src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp

```diff
@@ -1828,6 +1828,118 @@ TEST_P(CpuVaTensorsTests, checkResultsAfterStateTensorsUseImportCpuVa1) {
+
+    const auto shape = ov::PartialShape{1, 16, 16, 16};
+    auto model = createModel(element::f32, shape, "N...");
+    auto compiledModel = core->compile_model(model, target_device, configuration);
+
+    const auto elementType = compiledModel.inputs()[0].get_element_type();
+    const size_t byteSize = ov::shape_size(shape.get_shape()) * elementType.size();
+
+    for (size_t i = 0; i < 2; ++i) {
+        auto inferRequest1 = compiledModel.create_infer_request();
+        auto inferRequest2 = compiledModel.create_infer_request();
+
+        const auto alignment = i == 0 ? std::align_val_t(4096) : std::align_val_t(2);
+        float* data = static_cast<float*>(::operator new(byteSize, alignment));
+        auto tensor = ov::Tensor{elementType, shape.get_shape(), data};
+        OV_ASSERT_NO_THROW(inferRequest1.set_tensor(compiledModel.inputs()[0], tensor))
+        OV_ASSERT_NO_THROW(inferRequest2.set_tensor(compiledModel.inputs()[0], tensor))
+
+        auto remoteContextParams =
+            core->get_default_context("NPU")
+                .get_params();  // cannot get context from ZeroInitStructsHolder instance due to static member accross
+                                // different libraries (will differ from intel_npu::Plugin)
+        auto l0Context = remoteContextParams.find(intel_npu::l0_context.name());
+        EXPECT_TRUE(l0Context != remoteContextParams.end());
+        EXPECT_TRUE(::intel_npu::zeroUtils::memory_was_allocated_in_the_same_l0_context(
+            static_cast<ze_context_handle_t>(l0Context->second.as<void*>()),
+            data));
+
+        OV_ASSERT_NO_THROW(inferRequest1.infer());
+        OV_ASSERT_NO_THROW(inferRequest2.infer());
+
+        inferRequest1 = {};
+        inferRequest2 = {};
+        tensor = {};
+
+        EXPECT_FALSE(::intel_npu::zeroUtils::memory_was_allocated_in_the_same_l0_context(
+            static_cast<ze_context_handle_t>(l0Context->second.as<void*>()),
+            data));
+
+        ::operator delete(data, alignment);
+    }
+}
+
+TEST_P(CpuVaTensorsTests, SetAlignedDifferentTensorTwiceForSameInferRequestsNoThrow) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    const auto shape = ov::PartialShape{1, 16, 16, 16};
+    auto model = createModel(element::f32, shape, "N...");
+    auto compiledModel = core->compile_model(model, target_device, configuration);
+
+    const auto elementType = compiledModel.inputs()[0].get_element_type();
+    const size_t byteSize = ov::shape_size(shape.get_shape()) * elementType.size();
+
+    auto inferRequest = compiledModel.create_infer_request();
+    float* data = static_cast<float*>(::operator new(byteSize, std::align_val_t(4096)));
+    auto tensor1 = ov::Tensor{elementType, shape.get_shape(), data};
+    auto tensor2 = ov::Tensor{elementType, shape.get_shape(), data};
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor1));
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor2));
```

Comment:
I think we will skip the second set_tensor since the data ptr is same. But what if we change shape or stride of the tensor and set it again, do we still want to ignore the set?